Expensify/App
# Implemented Search View

@marcaaron @shawnborton 

### Details
Replaced `ChatSwitcherView` with a newly implemented `SearchView` component. 
It shows a list with recent reports, with or without any messages sent.
If a report is selected, the search closes and we get redirected to it.

Deleted everything related to `ChatSwitcherView`, that would not be used anymore.

### Fixed Issues
Fixes #1338 

### Tests
1. Click the search icon;
2. Verify the list appear, with recent reports that contain or not a message sent;
3. Verify that after pressing / clicking in one of the rows, the report chat comes up.

### Tested On


### Screenshots

#### Web
![web](https://i.ibb.co/bBfQ5FB/web.gif)

#### Mobile Web
![mobile web](https://i.ibb.co/tDYjnXW/mobile-web.gif)

#### Desktop
![desktop](https://i.ibb.co/6HHTZkp/desktop.gif)

#### iOS
![ios](https://i.ibb.co/84cFRcJ/ios.png)

#### Android
![android](https://i.ibb.co/dKRMsZp/android.gif)
